pkg/certificate/types.go

```diff
@@ -153,3 +156,16 @@ type MRCEventBroker interface {
 	// an error if the subscription goes awry.
 	Watch(context.Context) (<-chan MRCEvent, error)
 }
+
+// UpdateMRCStatus updates the status of an MRC to the desired state
+type UpdateMRCStatus func(ctx context.Context, mrc *v1alpha2.MeshRootCertificate) error
+
+// CheckMRCStatus checks if the MRC status is in the desired state
+type CheckMRCStatus func(ctx context.Context, mrc *v1alpha2.MeshRootCertificate) bool
+
+// MRCReconciler defines the actions of a reconciler used to determine when and how to update an MRC's status
+type MRCReconciler struct {
+	mrcName      string
```

Comment:
Shouldn't these fields be exported?

Reply:
I had envisioned only creating the `MRCReconciler` in the certificate package. Does it make more sense for these fields to be exported?

Reply:
prefer unexported if it's unused outside the package